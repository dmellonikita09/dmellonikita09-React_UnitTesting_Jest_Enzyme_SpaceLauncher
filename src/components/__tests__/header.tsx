import { shallow } from 'enzyme';
import React from 'react';
import Header from '../header';

describe('Header', () => {
  it('renders header with image and children', () => {
      const imgSrc = 'testImage.png';
      const wrapper = shallow(<Header image={imgSrc}><div id='testChildren'/></Header>);  
      const img = wrapper.find('Styled(img)');
      const children = wrapper.find('#testChildren');
      expect(img.prop("src")).toEqual(imgSrc);
      expect(children).toHaveLength(1);      
  });
  it('renders without error and matches snapshot',() =>{
    const component=shallow(<Header />);
    expect (component).toMatchSnapshot();
  });
});
