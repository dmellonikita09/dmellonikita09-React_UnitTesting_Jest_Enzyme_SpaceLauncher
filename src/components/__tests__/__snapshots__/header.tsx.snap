// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without error and matches snapshot 1`] = `
<Styled(div)>
  <Styled(img)
    alt="Space dog"
    round={true}
    src="dog-3.png"
  />
  <div>
    <h2>
      Space Explorer
    </h2>
    <Styled(h5)>
      ée
    </Styled(h5)>
  </div>
</Styled(div)>
`;
